update-carter-package-references/workflow.yml

```diff
@@ -0,0 +1,58 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
```

Comment:
In case it's not clear, the file here is just a template that is supposed to be copied in other repos.
We will not be running the workflow from this repository.